### PR TITLE
feat(coordinator): add coordinator controller

### PR DESCRIPTION
**Pull request by 0xmad**
# Description

- [x] Add zkeys download script
- [x] Add controller
- [x] Update proof generator service
- [x] Add e2e tests

## Additional Notes

Need to add guard and input checks

## Related issue(s)

Related to https://github.com/privacy-scaling-explorations/maci-rpgf/issues/60

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: cli/ts/index.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env node
```
Comment (ctrlc03) [inline cli/ts/index.ts:1]: why removing this? does it work without?

**Review by ctrlc03 — APPROVED** [review]
Great work thank you! just left one comment

File: cli/ts/index.ts (hunk 1 above)
Comment (0xmad) [reply cli/ts/index.ts:1]: Right, forgot to return it back.

File: coordinator/scripts/downloadZkeys.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import dotenv from "dotenv";
+import * as tar from "tar";
+
+import fs from "fs";
+import https from "https";
+import path from "path";
+
+dotenv.config({ path: [path.resolve(__dirname, "../.env"), path.resolve(__dirname, "../.env.example")] });
```
[the remark this answers is not in the document]
Comment (0xmad) [reply coordinator/scripts/downloadZkeys.ts:8]: User won't be able to run it without your own .env file. And what type of errors?

File: coordinator/tests/app.test.ts [hunk 3]
```diff
@@ -1,26 +1,235 @@
...
+      voteOptionTreeDepth: VOTE_OPTION_TREE_DEPTH,
+      messageBatchDepth: MSG_BATCH_DEPTH,
+      processMessagesZkeyPathNonQv: path.resolve(
+        __dirname,
+        "../zkeys/ProcessMessagesNonQv_10-2-1-2_test/ProcessMessagesNonQv_10-2-1-2_test.0.zkey",
+      ),
+      tallyVotesZkeyPathNonQv: path.resolve(
+        __dirname,
+        "../zkeys/TallyVotesNonQv_10-1-2_test/TallyVotesNonQv_10-1-2_test.0.zkey",
+      ),
```
[the remark this answers is not in the document]
Comment (0xmad) [reply coordinator/tests/app.test.ts:60]: QV is already tested for cli, doesn't matter what keys are tested here

File: coordinator/scripts/downloadZkeys.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kittybest) [reply coordinator/scripts/downloadZkeys.ts:8]: just wondering why adding .env.example here. thanks for the answer.

**Review by kittybest — APPROVED** [review]

**Pull request merged**